presentation/components/src/presentation-components/table/DataProvider.ts

```diff
@@ -306,9 +357,19 @@ const createRow = (descriptor: Readonly<Descriptor>, item: Readonly<Item>): RowI
 
   return {
     key,
-    cells: descriptor.fields.map((field): CellItem => ({
-      key: field.name,
-      record: ContentBuilder.createPropertyRecord({ field }, item).record,
-    })),
+    cells: descriptor.fields.map((field: Field): CellItem => {
+      const mergedCellsCount = mergedCellsCounts[field.name];
+      let nestedField: Field|undefined;
+      if ((mergedCellsCount ?? 0) > 1) {
+        nestedField = sameInstanceFields[field.name] ;
+        nestedField.name = field.name;
+      }
+      return {
+        key: field.name,
+        record: ContentBuilder.createPropertyRecord({field: nestedField ?? field}, item).record,
+        mergedCellsCount,

```

Comment:
I'd set these two attributes only if `mergedCellsCount !== undefined`. This would remove those empty changes from our snapshots.

Reply:
Updated

Reply:
This is very hard to read.. I'd rewrite like this:
```ts
  return {
    key,
    cells: descriptor.fields.map((field: Field): CellItem => {
      let itemProps: Partial<CellItem> = {};
      const mergedCellsCount = mergedCellsCounts[field.name];
      if (mergedCellsCount) {
        itemProps.mergedCellsCount = mergedCellsCount;
        itemProps.alignment = HorizontalAlignment.Center;

        const nestedField = sameInstanceFieldsMap[field.name];
        nestedField.name = field.name;
        field = nestedField;
      }
      return {
        key: field.name,
        record: ContentBuilder.createPropertyRecord({ field }, item).record,
        ...itemProps,
      };
    }),
  };
```

Reply:
Yeh. Looks a lot better, thanks!